ipn/ipnlocal/local.go

```diff
@@ -2918,6 +2920,11 @@ func (b *LocalBackend) EditPrefs(mp *ipn.MaskedPrefs) (ipn.PrefsView, error) {
 		b.logf("EditPrefs requests SSH, but disabled by envknob; returning error")
 		return ipn.PrefsView{}, errors.New("Tailscale SSH server administratively disabled.")
 	}
+	if p1.RunWebClient && !envknob.Bool("TS_DEBUG_WEB_UI") {
+		b.mu.Unlock()
```

Comment:
Kinda weird that we don't just defer the unlock above. Each of these cases manually unlock before returning.

Reply:
The docs for Unlock mention:

> It is a run-time error if m is not locked on entry to Unlock.

There is an inline comment later in this function that b.setPrefsLockedOnEntry does a b.mu.Unlock, so I suspect adding a defer to unlock again would cause an error.

Reply:
just tested, and adding a defer unlock definitely errors.

Reply:
Oooh I see. Wonder why `b.setPrefsLockedOnEntry` does an unlock. Feels like it's mixing levels a little. But this change lgtm